codice/ddf
# DDF-4269 Read/Write sharing permissions

Adds read-only permission to the sharing UI

#### Who is reviewing it? 
@Schachte 
@leo-sakh 
@gjvera 
@djblue 

#### Select relevant component teams: 
@codice/ui 
@codice/security 

#### Ask 2 committers to review/merge the PR and tag them here.
@andrewkfiedler
@bdeining

#### How should this be tested?

For both workspaces and search forms test each possible permission:
Group: none, read, write 
Individual: none, read, write, share

Ensure interaction menu (three dot) displays the correct options in each case (edit, delete, share)
Ensure saving works (or is not allowed in the case of readonly) 
etc. 
#### What are the relevant tickets?
[DDF-4269](https://codice.atlassian.net/browse/DDF-4269)
#### Screenshots
individual dropdown options
<img width="996" alt="screen shot 2018-10-25 at 1 38 32 pm" src="https://user-images.githubusercontent.com/37838624/47520073-2f2f7e80-d85d-11e8-94e4-fa79e4b3e465.png">
role dropdown options
<img width="856" alt="screen shot 2018-10-25 at 1 38 52 pm" src="https://user-images.githubusercontent.com/37838624/47520074-2f2f7e80-d85d-11e8-89c4-bd4d507c5b86.png">
warning when attempting to save a readonly workspace. We only block on save (instead of blocking any changes at all) because many actions (e.g. changing search criteria) count as an edit in the UI. 
<img width="827" alt="screen shot 2018-10-25 at 1 37 24 pm" src="https://user-images.githubusercontent.com/37838624/47520075-2f2f7e80-d85d-11e8-8fff-84acfe6e0685.png">

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
